src/utils/api/npg/index.ts

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable sonarjs/cognitive-complexity */
+/* eslint-disable sonarjs/no-identical-functions */
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { pipe } from "fp-ts/function";
+import { toError } from "fp-ts/lib/Either";
+import { SessionWalletCreateResponse } from "../../../../generated/definitions/webview-payment-wallet/SessionWalletCreateResponse";
+import { createClient as createWalletClient } from "../../../../generated/definitions/webview-payment-wallet/client";
+import { WalletId } from "../../../../generated/definitions/webview-payment-wallet/WalletId";
+import { WalletVerifyRequestsResponse } from "../../../../generated/definitions/webview-payment-wallet/WalletVerifyRequestsResponse";
+import { ErrorsType } from "../../errors/errorsModel";
+import { getConfigOrThrow } from "../../../config";
+import config from "../config";
+
+const NODE_ENV = getConfigOrThrow().WALLET_CONFIG_API_ENV;
+const WALLET_CONFIG_API_HOST = getConfigOrThrow().WALLET_CONFIG_API_HOST;
+const WALLET_CONFIG_API_BASEPATH =
+  getConfigOrThrow().WALLET_CONFIG_API_BASEPATH;
+
+const apiWalletClient = createWalletClient({
+  baseUrl: NODE_ENV === "development" ? "" : WALLET_CONFIG_API_HOST,
+  basePath: WALLET_CONFIG_API_BASEPATH as string,
+  fetchApi: config.fetchWithTimeout
+});
+
+const sessionsFields = async (
+  bearer: string,
+  walletId: WalletId,
+  onResponse: (data: SessionWalletCreateResponse) => void,
+  onError: (e: string) => void
+) =>
+  await pipe(
+    TE.tryCatch(
+      () =>
+        apiWalletClient.createSessionWallet({
+          walletId,
+          bearerAuth: bearer
+        }),
+      (_e) => toError
+    ),
+    TE.fold(
+      (err) => {
+        onError(ErrorsType.GENERIC_ERROR);
+        return TE.left(err);
+      },
+      (myResExt) => async () =>
+        pipe(
+          myResExt,
+          E.fold(
+            () => ({}),
```

Comment:
You could call the onError(ErrorsType.GENERIC_ERROR) method. So you could predict the opening of the error modal with a generic error.

Reply:
Thanks for the suggestion, I've updated the code following your instructions. This also allowed me to re enable the previously skipped test.
Note that errors handling is outside the scope of this pr, so I'm going to improve this part of the code with an upcoming pr.